internal/certs/certs.go

```diff
@@ -162,12 +168,17 @@ func newCA() (*x509.Certificate, crypto.PrivateKey, error) {
 		NotAfter:              time.Now().AddDate(0, 0, 365).UTC(),
 		KeyUsage:              x509.KeyUsageDigitalSignature | x509.KeyUsageKeyEncipherment | x509.KeyUsageCertSign,
 		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+		IsCA:                  true,
 		BasicConstraintsValid: true,
 	}
 
-	caPrivKey, err := rsa.GenerateKey(rand.Reader, 4096)
+	caBytes, err := x509.CreateCertificate(rand.Reader, ca, ca, &caPrivKey.PublicKey, caPrivKey)
```

Comment:
Previously we were only creating the `x509.Certificate` manually, which does not give us any way to serialize it (as far as I can tell).

By calling `CreateCertificate` it creates the asn.1 DER bytes we need to serialize the certificate. When doing this I ran into an error which I believe is because we were not setting `IsCA` above.

I left a TODO below about this, but I could not find any other way to allow us to serialize this CA.